test/new-e2e/pkg/environments/aws/host/host.go

```diff
@@ -156,6 +166,13 @@ func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.Ho
 			return err
 		}
 
+		if params.installDocker {
+			_, _, err := docker.NewManager(*awsEnv.CommonEnvironment, host, true)
+			if err != nil {
+				return err
+			}
+		}
+
```

Comment:
This seems to cause issues with a fresh VM, due to multiple things running apt at the same time. Still figuring out how to get around that:
```
         +  pulumi:pulumi:Stack REDACTED **creating failed (49s)** 1 error; 2 messages
        Diagnostics:
          pulumi:pulumi:Stack (REDACTED):
            fakeintake task private ip found: REDACTED
            Waiting for fakeintake at REDACTED to be healthy
            error: update failed
        
          command:remote:Command (remote-aws-vm-cmd-apt-install-docker.io-REDACTED):
            error: Process exited with status 100: running "DEBIAN_FRONTEND=\"noninteractive\" sudo apt-get install -y docker.io":
            E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 2147 (apt-get)
            E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?
```